cognite/client/data_classes/data_modeling/views.py

```diff
@@ -863,3 +866,88 @@ def dump(self, camel_case: bool = True) -> dict[str, Any]:
             output["connection_type"] = "multi_reverse_direct_relation"
 
         return output
+
+
+@dataclass
+class PropertyUnitReference:
+    property: str
+    unit: UnitReference | UnitSystemReference
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        return {"property": self.property, "unit": self.unit.dump(camel_case)}
+
+    @classmethod
+    def load(cls, data: dict) -> PropertyUnitReference:
+        return cls(
+            property=data["property"],
+            unit=UnitReference.load(data["unit"])
+            if "externalId" in data["unit"]
+            else UnitSystemReference.load(data["unit"]),
+        )
+
+
+@dataclass(frozen=True)
+class SourceDef(ViewId):
+    target_units: list[PropertyUnitReference] = field(default_factory=list)
+
+    def dump(self, camel_case: bool = True, include_type: bool = True) -> dict[str, Any]:
+        output: dict[str, Any] = {
+            "source": {
+                "space": self.space,
+                "externalId" if camel_case else "external_id": self.external_id,
+                "version": self.version,
+            }
+        }
+        if include_type:
+            output["source"]["type"] = "view"
+        if self.target_units:
+            output["targetUnits"] = [v.dump(camel_case) for v in self.target_units]
+        return output
+
+    @classmethod
+    def from_view_id(cls, view_id: ViewId, target_units: list[PropertyUnitReference] | None = None) -> SourceDef:
+        return cls(
+            space=view_id.space,
+            external_id=view_id.external_id,
+            version=view_id.version,
+            target_units=target_units or [],
+        )
+
+    @classmethod
+    def load(cls, data: dict | SourceDef | tuple[str, str] | tuple[str, str, str] | ViewId | View) -> SourceDef:
```

Comment:
Same question here, I think I forgot when we use private vs public load 😅

Reply:
We typically use private load when inheriting from `CogniteObject` which then expose a public load that assumes a private load method existing. But if you don't inherit from a `CogniteObject` I think it should be a public load method.

Reply:
Ah, right, makes sense 👌